add `warm_up_period` to `init`

djrtwo opened the pull request:
addresses #140 

- add `warm_up_period` param to `init`. Sets `START_EPOCH` and initial `current_epoch` to an epoch in the future and thus stops calls to `initialize_epoch` until then.
- add `WARM_UP_PERIOD` constant on the contract. I dont think we actually need this. Might want a method for validators to know if we are still in the warm up period. Will think about it in the morning.
- add explicit tests for `warm_up_period` functionality
- make all tests have a warm_up_period as default because that is how we expect to deploy. A lot of tests assumed we started at `epoch == 0` so I had to fix them to assume we start at `START_EPOCH`

ChihChengLiang commented on casper/contracts/simple_casper.v.py:
```diff
@@ -95,6 +95,7 @@
 
 # Length of an epoch in blocks
 EPOCH_LENGTH: public(int128)
+WARM_UP_PERIOD: public(int128)
```
would be helpful to comment `# warm up period in blocks` here

ChihChengLiang commented on tests/test_deposit.py:
```diff
@@ -84,3 +86,34 @@ def test_deposit_updates_total_deposits(casper, funded_privkey, deposit_amount,
+    expected_start_dynasty = casper.dynasty() + 2
+    assert casper.validators__start_dynasty(validator_index) == expected_start_dynasty
+
+    new_epoch()  # new_epoch mines through warm_up_period on first call
```
If I understand correctly, as described in #141 `initialize_epoch` can be called after the `WARM_UP_PERIOD` right? 
why `new_epoch`can be called before the expected start dynasty?

djrtwo replied:
`new_epoch` actually encapsulates walking through the warm up period if there is a warm up period.

This functionality of the `new_epoch` fixture it tested [here](https://github.com/ethereum/casper/pull/142/files#diff-d2d5f544b29602ba7340415d252e40b6R14) and then used throughout.

djrtwo commented:
- added a comment to `WARM_UP_PERIOD` being in blocks
- added a comment to `new_epoch` fixture noting that it walks through the warm up period if it exists.
